source/fundamentals/csfle.txt

```diff
@@ -0,0 +1,300 @@
+==================================
+Client-Side Field Level Encryption
+==================================
+
+.. default-domain:: mongodb
+
+Client-Side Field Level Encryption (CSFLE) allows you to encrypt
+specific data fields within a document with your MongoDB client application before sending the data to the server.
+Starting in MongoDB 4.2 Enterprise, you can perform this client-side encryption automatically.
+
+With CSFLE, your client application encrypts fields client-side without requiring any server-side configuration or
+directives. CSFLE is useful for situations in which applications must guarantee that unauthorized parties, including
+server administrators, cannot read the encrypted data.
+
+This guide is a quick introduction to CSFLE using the Java driver. For in-depth information on how CSFLE works, see
+the :manual:`CSFLE reference </core/security-client-side-encryption/>` documentation. For a real-world scenario and
+implementation, see our `CSFLE Guide <https://docs.mongodb.com/drivers/security/client-side-field-level-encryption-guide>`_.
+
+Installation
+------------
+
+To get started with CSFLE in your client application, you need
+
+- the MongoDB Java driver
+- ``libmongocrypt``
+- ``mongocryptd`` if using automatic encryption (Enterprise or Atlas)
+
+``libmongocrypt``
```

Comment:
Let's just refer to this as `mongodb-crypt`, since libmongocrypt.so/.dll is abstract from the user by the Java module of that name